sound/soc/sof/ipc4-topology.c

```diff
@@ -1112,7 +1113,15 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 			 * for all widgets with the same stream name
 			 */
 			i = 0;
+			node_id_shift = 0;
 			list_for_each_entry(w, &sdev->widget_list, list) {
+				if (w->id != snd_soc_dapm_dai_in &&
+				    w->id != snd_soc_dapm_dai_out)
+					continue;
+
+				if (w->use_count > 0 && strstr(w->widget->name, "ALH"))
+					node_id_shift++;
```

Comment:
I couldn't figure out what the use_count would be used for, and I don't get what the 'shift' means here.

Clearly I completely missed the idea behind this commit...

Reply:
> I couldn't figure out what the use_count would be used for, and I don't get what the 'shift' means here.
> 
> Clearly I completely missed the idea behind this commit...

As I don't know how to get group_id for the widget, so I tried to calculate the index and `node_id_shift` is to replace `group_id`. But @ranj063 has proved that calculating with use_count doesn't work. So that is not an option anymore